src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/AccessProfileServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+        Map<String, RoleToAccessProfileDefinition> roleToAccessProfileDefinitionMap =
+            toRoleNameAsKeyMap(roleToAccessProfiles);
+
+        for (RoleAssignment roleAssignment : filteringResults.getRoleAssignments()) {
+
+            RoleToAccessProfileDefinition roleToAccessProfileDefinition =
+                roleToAccessProfileDefinitionMap.get(roleAssignment.getRoleName());
+
+            if (roleToAccessProfileDefinition != null && !roleToAccessProfileDefinition.getDisabled()) {
+                List<String> authorisations = roleToAccessProfileDefinition.getAuthorisationList();
+                List<String> roleAssignmentAuthorisations = roleAssignment.getAuthorisations();
+
+                if (authorisationsAllowMappingToAccessProfiles(authorisations, roleAssignmentAuthorisations)) {
+                    accessProfiles.addAll(createAccessProfiles(roleAssignment, roleToAccessProfileDefinition));
+                }
+            }
+        }
+        return accessProfiles;
+    }
+
+    private Map<String, RoleToAccessProfileDefinition> toRoleNameAsKeyMap(
+        List<RoleToAccessProfileDefinition> roleToAccessProfiles) {
+        return roleToAccessProfiles
+            .stream()
+            .collect(Collectors.toMap(RoleToAccessProfileDefinition::getRoleName,
+                                      Function.identity()));
+    }
+
+    private boolean authorisationsAllowMappingToAccessProfiles(List<String> authorisations,
+                                                               List<String> roleAssignmentAuthorisations) {
+        if (roleAssignmentAuthorisations != null
+            && authorisations.size() > 0) {
+            Collection<String> filterAuthorisations = CollectionUtils
+                .intersection(roleAssignmentAuthorisations, authorisations);
+
+            return !filterAuthorisations.isEmpty();
+        }
+        return authorisations.size() == 0;
+    }
+
+    private List<AccessProfile> createAccessProfiles(RoleAssignment roleAssignment,
+                                                     RoleToAccessProfileDefinition roleToAccessProfileDefinition) {
+        List<String> accessProfileList = roleToAccessProfileDefinition.getAccessProfileList();
+        return accessProfileList
+            .stream()
+            .map(accessProfileValue -> AccessProfile.builder()
+                .accessProfile(accessProfileValue)
+                .securityClassification(roleAssignment.getClassification())
+                .readOnly(readOnly(roleAssignment, roleToAccessProfileDefinition))
+                .build()).collect(Collectors.toList());
+    }
+
+    private Boolean readOnly(RoleAssignment roleAssignment,
+                             RoleToAccessProfileDefinition roleToAccessProfileDefinition) {
+        return BooleanUtils.isTrue(roleAssignment.getReadOnly()) ||
+            BooleanUtils.isTrue(roleToAccessProfileDefinition.getReadOnly());
+    }
```

Comment:
Any reason we can't simplify to primitive?
```suggestion
    private boolean readOnly(RoleAssignment roleAssignment,
                             RoleToAccessProfileDefinition roleToAccessProfileDefinition) {
        return roleAssignment.getReadOnly() || roleToAccessProfileDefinition.getReadOnly();
    }
```

Reply:
it might throw null pointer exception